### PR TITLE
pull: support pulling git-only imports

### LINKED ISSUE #3377: Failure to pull after importing a git repo
**Please provide information about your setup**
DVC version(i.e. `dvc --version`), Platform and method of installation (pip, homebrew, pkg Mac, exe (Windows), DEB(Linux), RPM(Linux))

DVC 0.86.4, conda installation, Windows. 

See https://discuss.dvc.org/t/importing-from-a-git-repo-then-pulling/320 for background. 

After `import`ing a directory from a git repository (not a dvc repository), `dvc pull` fails with `ERROR: unexpected error - 'ExternalGitRepo' object has no attribute 'cache'`.

### PR DESCRIPTION
**Pull request by skshetry**
* [x] ❗ I have followed the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) checklist.

* [ ] 📖 If this PR requires [documentation](https://dvc.org/doc) updates, I have created a separate PR (or issue, at least) in [dvc.org](https://github.com/iterative/dvc.org) and linked it here. If the CLI API is changed, I have updated [tab completion scripts](https://github.com/iterative/dvc/tree/master/scripts/completion).

* [x] ❌ I will check DeepSource, CodeClimate, and other sanity checks below. (We consider them recommendatory and don't expect everything to be addressed. Please fix things that actually improve code or fix bugs.)

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

Fix #3377.

### CODE REVIEW

File: dvc/repo/fetch.py [hunk 1]
```diff
@@ -74,8 +74,12 @@ def _fetch_external(self, repo_url, repo_rev, files, jobs):
     failed = 0
     try:
         with external_repo(repo_url, repo_rev) as repo:
-            repo.cache.local.cache_dir = self.cache.local.cache_dir
+            if not hasattr(repo, "cache"):
+                return _fetch_external_git(
+                    self.cache.local, repo.root_dir, files
+                )
```
Comment (efiop) [inline dvc/repo/fetch.py]:
```suggestion
            if not isinstance(repo, ExternalRepo):
                return 0, 0
```

File: dvc/repo/fetch.py [hunk 2]
```diff
@@ -74,8 +74,12 @@ def _fetch_external(self, repo_url, repo_rev, files, jobs):
...
```
Comment (efiop) [inline dvc/repo/fetch.py]: For the record: discussed with @skshetry that we might be missing support git-files imported from dvc-repo.

File: tests/func/test_data_cloud.py [hunk 3]
```diff
@@ -709,3 +711,56 @@ def test_verify_checksums(tmp_dir, scm, dvc, mocker, tmp_path_factory):
...
+
+    for item in ["foo", "new_dir", dvc.cache.local.cache_dir]:
+        remove(item)
+    os.makedirs(dvc.cache.local.cache_dir, exist_ok=True)
+    clean_repos()
+
+    assert dvc.pull(force=True)["downloaded"] == 2
+
+    assert os.path.exists("foo")
+    assert open("foo").read() == "foo"
```
Comment (efiop) [inline tests/func/test_data_cloud.py]: leaking fds, not good :) You could use `(repo_dir / "foo").read_text() == "foo"` instead. Same with `exists`.

Comment (skshetry) [reply tests/func/test_data_cloud.py]: Fixed. Thanks.

**Comment by efiop** [comment]
Thank you @skshetry ! :pray:

**Pull request merged**